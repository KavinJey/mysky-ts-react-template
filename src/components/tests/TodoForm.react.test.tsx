import { createStore, StoreProvider } from "easy-peasy";
import renderer from "react-test-renderer";
import TodoForm from "../TodoForm";
import { initialState, StoreModel } from "../../state/store";

test("Check if forms renders correctly", () => {
  const mockStore = createStore<StoreModel>(initialState);
  const app = (
    <StoreProvider store={mockStore}>
      <TodoForm />
    </StoreProvider>
  );

  const renderedComponent = renderer.create(app);
  let tree = renderedComponent.toJSON();
  expect(tree).toMatchSnapshot();
});
