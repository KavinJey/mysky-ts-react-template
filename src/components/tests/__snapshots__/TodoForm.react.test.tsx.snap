// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if forms renders correctly 1`] = `
Array [
  <div
    className="ui labeled input"
  >
    <div
      className="ui label label"
      onClick={[Function]}
    >
      todo-text
    </div>
    <input
      onChange={[Function]}
      placeholder="Todo item..."
      type="text"
      value=""
    />
  </div>,
  <button
    className="ui button"
    onClick={[Function]}
    style={
      Object {
        "marginLeft": "10px",
      }
    }
  >
    Add Todo
  </button>,
]
`;
